modular_skyrat/modules/verbs/code/subtle.dm

```diff
@@ -51,17 +42,18 @@
 
 	subtle_message = span_emote("<b>[user]</b>[space]<i>[user.say_emphasis(subtle_message)]</i>")
 
-	for(var/mob/ghosts in GLOB.dead_mob_list)
-		if(!ghosts.client || isnewplayer(ghosts))
+	var/list/viewers = get_hearers_in_view(1, user)
+
+	for(var/mob/ghost in GLOB.dead_mob_list)
+		if(!ghost.client || isnewplayer(ghost))
 			continue
-		var/their_turf = get_turf(src)
-		if(ghosts.stat == DEAD && ghosts.client && (ghosts.client.prefs.chat_toggles & CHAT_GHOSTSIGHT) && !(ghosts in viewers(their_turf, null)))
-			ghosts.show_message(subtle_message)
+		if(ghost.stat == DEAD && ghost.client && (ghost.client.prefs.chat_toggles & CHAT_GHOSTSIGHT) && !(ghost in viewers))
```

Comment:
```suggestion
		if(ghost.stat == DEAD && (ghost.client?.prefs.chat_toggles & CHAT_GHOSTSIGHT) && !(ghost in viewers))
```
can a ghost have a stat other than dead?

Reply:
Not sure, but it was in the original code-

Reply:
With how some of that old code was looking, that's by no means a guarantee of correctness lol. Worth a look into

Reply:
Stat is slated to be moved from `mob` to `mob/living`, so I'm removing the check anyways!